server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -910,7 +910,7 @@ async def delete_tsdb_records(project: str, uids: list[str]):
                     project=project
                 ),
             )
-            tsdb_connector.delete_tsdb_records(endpoint_ids=uids)
+            run_in_threadpool(tsdb_connector.delete_tsdb_records, endpoint_ids=uids)
```

Comment:
I think that it will just create a coroutine object without running it, you might need to add `await` 
```suggestion
           await run_in_threadpool(tsdb_connector.delete_tsdb_records, endpoint_ids=uids)
```